crates/router/src/macros.rs

```diff
@@ -51,3 +51,18 @@ macro_rules! async_spawn {
         tokio::spawn(async move { $t });
     };
 }
+
+#[macro_export]
+macro_rules! get_keys_of_option_nones {
+    [$(($key:expr, $option:expr)), *] => {
+        {
+            let mut keys = Vec::new();
```

Comment:
```suggestion
            let mut keys: Vec<&str> = Vec::new();
```
for better error reporting.